crates/symbolicator/src/services/download/gcs.rs

```diff
@@ -300,6 +300,84 @@ impl GcsDownloader {
         }
     }
 
+    /// Uploads a file to GCS.
+    pub async fn upload(
+        &self,
+        destination: GcsRemoteDif,
+        contents: &[u8],
+    ) -> Result<DownloadStatus, DownloadError> {
```

Comment:
hah, i assume the naming on this is one of the awkward quirks that comes from reusing what's already present to add in upload functionality.

as a small nit, i would like to ask that before this is merged that a second sweep over the logs to is performed to at least ensure that those match the actual action here: the logs right now say things like "insufficient permissions to download x" when it should be "insufficient permissions to upload x".

Reply:
oh absolutely, the logs here need updating for sure! And yes, reusing all the types that have "download" in their name for uploads is probably the biggest gripe I have with the solution so far. Though I also don’t have any better ideas at the moment.